### ISSUE TITLE
Can librosa not be a strict dependency?

### ISSUE THREAD

**Issue by turian** [csteinmetz1/auraloss#36]
Given different versions of OSes with different versions of librosa, librosa is a huge pain to install on certain machines.

For example, I use lambdalabs GPUs and the librosa is wrong. This old librosa depends upon a particular pinned version of numba that's a huge pain to install:

```
ImportError: Numba needs NumPy 1.22 or less
```

Thus, I think librosa should be deprecated.

**Comment by csteinmetz1** [comment]
Hey @turian. Thanks for raising this. I think we add librosa as an optional dependency since its only needed for the mel filterbanks. Is scipy also a problem for you? I think we could move to be optional as well potentially.

**Comment by csteinmetz1** [comment]
If you want to take a look, I have started a [branch](https://github.com/csteinmetz1/auraloss/tree/librosa) that moves librosa and scipy to optional install by default.

**Comment by turian** [comment]
scipy is generally fine. Just librosa with it's weird numba pin sucks so much.

**Comment by turian** [comment]
I am happy the beta test when the branch is ready!

**Comment by csteinmetz1** [comment]
Great, could you give the [librosa](https://github.com/csteinmetz1/auraloss/tree/librosa) branch a try? I did some basic testing and things seemed to be working for me.

**Comment by turian** [comment]
```
pip3.8  install --upgrade git+https://github.com/csteinmetz1/auraloss.git@librosa
```

appears to work. If you submit a PR, I can also code review the diff if you like

**Comment by csteinmetz1** [comment]
Thanks, see #37.

**Comment by csteinmetz1** [comment]
Should be solved with #37 now and merged to main.
